apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/WriteTests.cs

```diff
@@ -0,0 +1,183 @@
+        }
+
+        [Fact]
+        public async Task WriteValues()
+        {
+            var testTimestamp = new DateTime(2017, 3, 17, 15, 30, 0);
+            var testDate = new DateTime(2017, 5, 9);
+            bool?[] bArray = {true, null, false};
+            long?[] lArray = {0, null, 1};
+            double?[] dArray = {0.0, null, 2.0};
+            string[] sArray = {"abc", null, "123"};
+            string[] bArrayArray =
+            {
+                Convert.ToBase64String(new byte[] {0, 1, 2}), null,
+                Convert.ToBase64String(new byte[] {1, 2, 3}),
+            };
+            DateTime?[] dtArray = new DateTime?[]
+                {new DateTime(2017, 3, 17), null, new DateTime(2017, 5, 9),};
+            DateTime?[] tmArray = new DateTime?[]
+                {new DateTime(2017, 3, 17, 5, 30, 0), null, new DateTime(2017, 5, 9, 12, 45, 0),};
+
+            Assert.Equal(await InsertAsync(new SpannerParameterCollection
+                {
+                    new SpannerParameter("BoolValue", SpannerDbType.Bool, true),
+                    new SpannerParameter("Int64Value", SpannerDbType.Int64, 1),
+                    new SpannerParameter("Float64Value", SpannerDbType.Float64, 2.0),
+                    new SpannerParameter("StringValue", SpannerDbType.String, "abc"),
+                    new SpannerParameter("TimestampValue", SpannerDbType.Timestamp, testTimestamp),
+                    new SpannerParameter("DateValue", SpannerDbType.Date, testDate),
+                    new SpannerParameter("BoolArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Bool),
+                        bArray),
+                    new SpannerParameter("Int64ArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Int64),
+                        lArray),
+                    new SpannerParameter("Float64ArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Float64),
+                        dArray),
+                    new SpannerParameter("StringArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.String),
+                        sArray),
+                    new SpannerParameter("BytesArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Bytes),
+                        bArrayArray),
+                    new SpannerParameter("TimestampArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Timestamp),
+                        tmArray),
+                    new SpannerParameter("DateArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Date),
+                        dtArray),
+                }),
+                1);
+            using (var reader = await GetLastRowAsync())
+            {
+                Assert.True(reader.GetFieldValue<bool>(reader.GetOrdinal("BoolValue")));
+                Assert.Equal(reader.GetFieldValue<long>(reader.GetOrdinal("Int64Value")), 1);
+                Assert.True(
```

Comment:
doesn't exist - there is a version that takes in a decimal.  I guess I could use zero decimal precision.  I'm fine with it the way it is though.